### PR TITLE
dts: arm: st: f4: fix die_temp channel

### PR DESCRIPTION
**Pull request by yishai1999**
The STM32F4 socs have different channels for the temperature sensor. Some are at channel 16 and some at channel 18. Made changes wherever it was relevant.
In short, the base configuration is to channel 16 and wherever it is supposed to be 18 it is overridden.

### CODE REVIEW

**Review by gautierg-st — COMMENTED** [review]
From what I gathered from the various F4 Ref Man (8 of them):
- F405/407/415/417 use channel 16 exclusively
- F412/413/423/427/429/437/439/446/469/479 use channel 18 exclusively
- F401/410 can use either one
- Not clear if F411 can use both or only 18 (channel 16 is sometimes mentioned, sometimes not)

So I'd say we can keep channel 18 by default, but set channel 16 in stm32f405.dtsi, and reset it to 18 in stm32f427.dtsi (which is already done). All other redefinitions could be removed.

What do you think?

**Comment by yishai1999** [comment]
> From what I gathered from the various F4 Ref Man (8 of them):
> 
> * F405/407/415/417 use channel 16 exclusively
> * F412/413/423/427/429/437/439/446/469/479 use channel 18 exclusively
> * F401/410 can use either one
> * Not clear if F411 can use both or only 18 (channel 16 is sometimes mentioned, sometimes not)
> 
> So I'd say we can keep channel 18 by default, but set channel 16 in stm32f405.dtsi, and reset it to 18 in stm32f427.dtsi (which is already done). All other redefinitions could be removed.
> 
> What do you think?

I went according to `modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_ll_adc.h` (lines 642-647).
According to that:
* 405/415/407/417/401/410 - Use channel 16
* 411/412/413/423/427/437/429/439/446/469/479 - Use channel 18
I then looked at the "inclusion tree" starting at `stm32f4.dtsi`. The only ones who include it are `stm32f401.dtsi` and `stm32f410.dtsi` which is why I thought the base `stm32f4.dtsi` should be configured to channel 16. I then kept going through the "inclusion tree" and changed whatver was needed which happened to be only `stm32f412.dtsi`!
When I saw there was only one file that needed to be updated I was kind of surprised that it was so easy so I looked at the commits and saw that it really used to be configured to channel 16 in `stm32f4.dtsi` until https://github.com/zephyrproject-rtos/zephyr/commit/ac7f2dad4ecba95ace2bc8473ff630e7707bf171.

In conclusion, unless I misunderstood something I think this is the correct way to have it.

**Review by gautierg-st — APPROVED** [review]
Let's keep the same channels as the ones in the LL, I agree with that.

**Review by erwango — APPROVED** [review]
